core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TableSink.java

```diff
@@ -53,10 +54,51 @@ public TableSink(TableSinkConfig tableSinkConfig) {
   }
 
   @Override
-  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
     super.configurePipeline(pipelineConfigurer);
     Preconditions.checkArgument(!Strings.isNullOrEmpty(tableSinkConfig.getRowField()),
                                 "Row field must be given as a property.");
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    if (inputSchema != null) {
+      validateInputAndOutputSchemaFields(inputSchema);
+
+      boolean rowFieldPresent = false;
+      for (Schema.Field field : inputSchema.getFields()) {
+        // simple type check for fields
+        if (!field.getSchema().getType().isSimpleType()) {
```

Comment:
a nullable string is represented as a union of null a string, which will fail this check. You can use `field.getSchema().isSimpleOrNullableSimple()`